src/ngx_http_lua_common.h

```diff
@@ -155,6 +158,8 @@ struct ngx_http_lua_main_conf_s {
     ngx_http_lua_conf_handler_pt    init_worker_handler;
     ngx_str_t                       init_worker_src;
 
+    ngx_http_lua_balancer_peer_data_t  *balancer_peer;
```

Comment:
Better add a comment explaining a bit why it is safe to store the balancer peer data in the main conf.